### PR TITLE
Split Fleece.h and Fleece.hh into multiple headers

### PR DESCRIPTION
**Pull request by snej**
Fleece.h is now an umbrella header that #includes the new headers:
  FLBase.h, FLCollections.h, FLDeepIterator.h, FLDoc.h, FLEncoder.h,
  FLJSON.h, FLKeyPath.h, FLMutable.h, FLValue.h

There is also FLExpert.h, which is stuff that clients are unlikely to
use, or too dangerous, or semi-deprecated.

On the C++ side I’ve only split out the expert stuff, into Expert.hh.

Also I renamed Base.h to CompilerSupport.h, to keep it from getting
mixed up with FLBase.h (not to mention Base.hh in LiteCore.)

### CODE REVIEW

**Comment by borrrden** [comment]
So everything that comes from the umbrella header should be exported as part of CBL for C?

File: API/fleece/Expert.hh [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+#include "Fleece.hh"
+#endif
+
+#include "FLExpert.h"
+
+FL_ASSUME_NONNULL_BEGIN
+
+namespace fleece {
+
+    //  Rarely-needed or advanced functionality; C++ wrapper around fleece/Obscure.h
```
Comment (borrrden) [inline API/fleece/Expert.hh:26]: Wrapped around what?  Should this be FLExpert.h ?

**Review by borrrden — CHANGES_REQUESTED** [review]
Just a comment mistake I think.

File: API/fleece/Expert.hh (hunk 1 above)
Comment (snej) [reply API/fleece/Expert.hh:26]: oops, yeah, it was briefly called Obscure.h before I went with "expert".

**Review by borrrden — APPROVED** [review]

**Review by pasin — APPROVED** [review]

**Pull request merged**